api/tests/authentication_tests.py

```diff
@@ -213,13 +224,63 @@ def test_get_fxa_uid_from_oauth_token_not_active_uses_cached_response_returns_er
             return
         self.fail("Should have raised AuthenticationFailed")
 
+    @responses.activate
+    def test_get_fxa_uid_from_oauth_token_has_wrong_scope_returns_error_response(  # noqa: E501
+        self,
+    ) -> None:
+        now_time = int(datetime.now().timestamp())
+        # Note: FXA iat and exp are timestamps in *milliseconds*
+        exp_time = (now_time + 60 * 60) * 1000
+        json_data: FxaResponse = {
+            "active": True,
+            "sub": self.uid,
+            "exp": exp_time,
+            "scope": "foo",
+        }
+        status_code = 200
+        fxa_response = _setup_fxa_response(status_code, json_data)
+        missing_scopes_token = "missing-scopes-123"
+        cache_key = get_cache_key(missing_scopes_token)
+
+        assert cache.get(cache_key) is None
+
+        # get fxa response with no fxa uid for the first time
+        try:
+            get_fxa_uid_from_oauth_token(missing_scopes_token)
+        except AuthenticationFailed as e:
+            assert (
+                str(e.detail)
+                == "FXA missing is missing scope: https://identity.mozilla.com/apps/relay."
+            )
+            assert responses.assert_call_count(self.fxa_verify_path, 1) is True
+        assert cache.get(cache_key) == fxa_response
+
+        # now check that the 2nd call did NOT make another fxa request
+        try:
+            get_fxa_uid_from_oauth_token(missing_scopes_token)
+        except AuthenticationFailed as e:
+            assert (
+                str(e.detail)
+                == "FXA missing is missing scope: https://identity.mozilla.com/apps/relay."
+            )
+            assert responses.assert_call_count(self.fxa_verify_path, 1) is True
+            return
+        self.fail("Should have raised AuthenticationFailed")
```

Comment:
question (non-blocking): do you want to test the caching behavior here too? Should we let the other tests test the cache and keep this test focused on the scope error? Or is there something about the scope error that we should test the cache here too?

Reply:
I was just following a pattern where most of the other tests in this file assert that the cache is working. I'll remove.